code/_onclick/hud/new_player.dm

```diff
@@ -348,14 +414,97 @@
+	addtimer(CALLBACK(src, PROC_REF(set_button_status), TRUE), (2 * SHUTTER_MOVEMENT_DURATION + SHUTTER_WAIT_DURATION), TIMER_CLIENT_TIME)
+
+	//get the shutter object used by our hud
+	var/atom/movable/screen/lobby/shutter/menu_shutter = locate(/atom/movable/screen/lobby/shutter) in hud.static_inventory
+
+	addtimer(CALLBACK(src, PROC_REF(toggle_menu), new_player, our_hud), SHUTTER_MOVEMENT_DURATION + 1, TIMER_CLIENT_TIME) //ever so slightly after the shutter pulls down
+	//animate the shutter
+	menu_shutter.setup_shutter_animation()
+	//animate bottom buttons' movement
+	if(our_hud.menu_hud_status)
+		collapse_buttons()
+	else
+		expand_buttons()
+
+///Moves the button to the top of the screen, leaving only the screen part in view
+///Sends a signal on the hud for the bottom buttons to listen to
+/atom/movable/screen/lobby/button/collapse/proc/collapse_buttons()
+	SEND_SIGNAL(hud, COMSIG_HUD_LOBBY_COLLAPSED)
+	var/matrix/transformation = matrix()
+	//wait for the shutter to come down
+	animate(src, transform = transformation, time = SHUTTER_MOVEMENT_DURATION + SHUTTER_WAIT_DURATION)
+	//then pull the button up with the shutter and leave it on the edge of the screen
+	animate(transform = transformation.Translate(x = 0, y = 134), time = SHUTTER_MOVEMENT_DURATION, easing = CUBIC_EASING|EASE_IN)
+
+///Extends the button back to its usual spot
+///Sends a signal on the hud for the bottom buttons to listen to
+/atom/movable/screen/lobby/button/collapse/proc/expand_buttons()
+	SEND_SIGNAL(hud, COMSIG_HUD_LOBBY_EXPANDED)
+	animate(src, transform = matrix(), time = SHUTTER_MOVEMENT_DURATION, easing = CUBIC_EASING|EASE_OUT)
+
+///Disables/enables menu buttons on the client's screen
+/atom/movable/screen/lobby/button/collapse/proc/toggle_menu(mob/dead/new_player/new_player, datum/hud/new_player/our_hud)
+	if(our_hud.menu_hud_status)
+		new_player.client.screen -= our_hud.toggleable_elements
+	else
+		new_player.client.screen += our_hud.toggleable_elements
+	our_hud.menu_hud_status = !our_hud.menu_hud_status
+
+/atom/movable/screen/lobby/shutter
+	icon = 'icons/hud/lobby/shutter.dmi'
+	icon_state = "shutter"
+	base_icon_state = "shutter"
+	screen_loc = "TOP:+143,CENTER:-73" //"home" position is off-screen
+	layer = LOBBY_SHUTTER_LAYER
+	always_shown = TRUE
+
+///Sets up the shutter pulling down and up. It's the same animation for both collapsing and expanding the menu.
+/atom/movable/screen/lobby/shutter/proc/setup_shutter_animation()
+	var/matrix/transformation = matrix()
+	transformation.Translate(x = 0, y = -143)
+	//bring down the shutter
+	animate(src, transform = transformation, time = SHUTTER_MOVEMENT_DURATION, easing = CUBIC_EASING|EASE_OUT)
+
+	//wait a little bit before bringing the shutter up
+	animate(transform = transformation, time = SHUTTER_WAIT_DURATION)
+
+	//pull the shutter back off-screen
+	animate(transform = matrix(), time = SHUTTER_MOVEMENT_DURATION, easing = CUBIC_EASING|EASE_IN)
```

Comment:
similar, why is there a second var here

Reply:
https://github.com/tgstation/tgstation/pull/76443/commits/9735e710e838bb6be51edc9c03dbe0e9688a8a48